src/query/plan/operator.cpp

```diff
@@ -438,7 +438,7 @@ class ScanAllCursor : public Cursor {
 
     AbortCheck(context);
 
-    while (!vertices_ || vertices_it_.value() == vertices_.value().end()) {
+    while (!vertices_ || vertices_it_.value() == vertices_end_it_.value()) {
```

Comment:
Does this really make a difference?

Reply:
Yes `end()` does a bunch of work to build that iterator.
Iterators are meant to be cheap, but this one is expensive.